app/controllers/hq/partners_controller.rb

```diff
@@ -5,4 +5,48 @@ class Hq::PartnersController < ApplicationController
   def index
     @partners= Partner.all
   end
+
+  def edit
+    load_partner
+    load_associations
+  end
+
+  def update
+    load_partner
+    load_associations
+    build_partner
+    save_partner or render 'edit'
+  end
+
+private
+  def load_partner
+    @partner = Partner.find(params[:id])
+  end
+
+  def load_associations
+    @provinces = Province.all
+    @statuses = Status.all
+  end
+
+  def build_partner
+    @partner ||= Partner.build
+    @partner.attributes = partner_params
+  end
+
+  def save_partner
+    if @partner.save
+      redirect_to hq_partner_url(@partner)
+    end
+  end
+
+  def partner_params
+    partner_params = params[:partner]
+    if partner_params
+      partner_params.permit(:name, :region, :contact_details, :province_id,
+                            :status_id, :start_date)
+    else
+      {}
```

Comment:
Skirting the aggregate issue for the moment because I would like to discuss that in more detail elsewhere.
No I don't think it is. We do the same things again and again in all of our controllers. We load, save and build our model ready for saving. 
Building is the same concept for the new, create and update methods. The only difference is their state. So for new we build a model without an associated database record and empty params, the same for create but we wan't to merge in form params, with an update we are looking at merging in form params to an existing record. This method covers all bases and is DRY.
 - build a new partner if one doesn't exist
 - merge in attributes if there are any
additionally Partner.new and Partner.build are just aliases https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation.rb#L124